content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,291 @@
+---
+Title: "How to Use GraphQL With Django"
+Description: "A guide on how to use GraphQL with Django."
+DatePublished: "2022-06-05"
+Categories:
+  - "python"
+  - "web-development"
+  - "computer-science"
+Tags:
+  - "APIs"
+  - "Django"
+  - "Queries"
+CatalogContent:
+  - "python-for-programmers"
+  - "paths/full-stack-engineer-career-path"
+---
+
+_**Prerequisites:** Understanding of Python and Django._  
+_**Versions:** Python 3.8.10, Django 4.0.4_
+
+## Introduction
+
+[GraphQL](https://www.codecademy.com/resources/docs/general/graphql) is a query language for [APIs](https://www.codecademy.com/resources/docs/general/api) and a runtime for fulfilling those queries with your existing data. Unlike a REST API, GraphQL APIs do not require verbs (`PUT`, `POST`, `GET`, etc.) for requests, nor do they need multiple endpoints. They have just one endpoint and making a query to that endpoint is all that's needed. 
+
+This tutorial will cover the creation of a CRUD (create, read, update, delete) GraphQL API for a restaurant with Django.
+
+## Properties of GraphQL
+
+The following terms would be used a lot when interacting with graphql. It is important that you know them, though we wouldn't cover all of them in this tutorial.
+
+- **Schema**: Describes the functionality available to the client applications that connect to it.
+- **Query**: A schema type that represents the `GET` request and defines the operations that can be used for reading or fetching data.
+- **Nesting**: Queries can be nested inside of other queries.
+- **Mutation**: A schema type that defines the kind of operations that can be done to modify data.
+- **Subscription**: Notifies the client server in real time about updates to the data.
+- **Resolvers**: Functions that return values for fields associated with existing schema types.
+
+## Setting up GraphQL with Django
+
+First, we need to install Django and create a project. I will be using a [bash script](https://github.com/Smyja/Django-bash) I created to do this.
+
+
+To use GraphQL with Django, you will need to install the `graphene-django` package:
+
+`pip install graphene-django`
+
+Add it to your `settings.py` file:
+
+```py
+INSTALLED_APPS = [
+    'graphene_django',
+
+    # ...
+]
```

Comment:
```suggestion
INSTALLED_APPS = [
  'graphene_django',

  # ...
]
```

Reply:
This is what i mean about changing the indent from 4 spaces to 2 spaces

Reply:
Oh, okay.